### PR TITLE
fix version incoherence crash

### PR DESCRIPTION
**Pull request by jdelga**
Fix a crash when the app has a `compose:material` version higher than the tweaks library.

It can be tested using a higher version in the module `app` than in `library`

                                               java.lang.NoClassDefFoundError: Failed resolution of: Landroidx/compose/material/TouchTargetKt;
                                               	at com.telefonica.tweaks.ui.TweakComponentsKt$EditableBooleanTweakEntryBody$2.invoke(TweakComponents.kt:326)
                                               	at com.telefonica.tweaks.ui.TweakComponentsKt$EditableBooleanTweakEntryBody$2.invoke(TweakComponents.kt:325)
                                               	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:116)
                                               	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
                                               	at com.telefonica.tweaks.ui.TweakComponentsKt.TweakRow(TweakComponents.kt:453)
                                               	at com.telefonica.tweaks.ui.TweakComponentsKt.EditableBooleanTweakEntryBody(TweakComponents.kt:317)
                                               	at com.telefonica.tweaks.ui.TweakComponentsKt$TweakGroupBody$1.invoke(TweakComponents.kt:179)
                                               	at com.telefonica.tweaks.ui.TweakComponentsKt$TweakGroupBody$1.invoke(TweakComponents.kt:160)
                                               	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:107)
                                               	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)

Before:

[tweaks_crash.webm](https://github.com/Telefonica/tweaks/assets/36664452/07598d11-d2c9-4f2a-abe7-1a5565aabd63)

After:

[tweaks_crash_fixed.webm](https://github.com/Telefonica/tweaks/assets/36664452/a2c4a986-1470-4375-8f0d-357ace7bfb19)

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -59,7 +59,7 @@ dependencies {
     implementation 'androidx.appcompat:appcompat:1.5.1'
     implementation 'com.google.android.material:material:1.7.0'
     implementation "androidx.compose.ui:ui:$compose_version"
-    implementation "androidx.compose.material:material:1.3.1"
+    implementation "androidx.compose.material:material:1.4.0"
```
Comment (pmartinbTEF) [inline app/build.gradle]: Could be extracted to a variable so it can be used for both app and library?

**Review by pmartinbTEF — COMMENTED** [review]
Nice!

File: app/build.gradle (hunk 1 above)
Comment (jdelga) [reply app/build.gradle]: Done! However, the issue happens when there is a different version between the library and one of our apps.

**Review by pmartinbTEF — APPROVED** [review]
Good job!

**Review by DevPabloGarcia — APPROVED** [review]
😄

**Pull request merged**